src/backend/parser/parse_expr.c

```diff
@@ -2278,7 +2278,10 @@ transformCoalesceExpr(ParseState *pstate, CoalesceExpr *c)
 		newargs = lappend(newargs, newe);
 	}
 
-	newc->coalescetype = select_common_type(pstate, newargs, "COALESCE", NULL);
+	if (sql_dialect == SQL_DIALECT_TSQL && select_common_type_hook && c->tsql_is_null)
```

Comment:
Are we continuing to use coalesce code path and handle the peculiarities of isnull separately anywhere we behave differently from coalesce?

Reply:
yes, we are not redesigning this ISNULL. Will continue using coalesce with the help of hook